File: internal/model/permission.go
package model

import "time"

// DiskPermission 智能体权限表
type DiskPermission struct {
	ID           uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID       string    `gorm:"index;size:64;not null" json:"userId"`
	AgentID      string    `gorm:"index:idx_agent;size:64;not null;default:''" json:"agentId"`
	AgentGroupID string    `gorm:"index:idx_agent;size:64;not null;default:''" json:"agentGroupId"`
	ResourceID   uint64    `gorm:"index;not null;default:0" json:"resourceId"`
	ResType      string    `gorm:"size:16;not null;default:''" json:"resType"`
	ResourcePath string    `gorm:"size:1024;not null;default:''" json:"resourcePath"`
	Permission   string    `gorm:"size:16;not null" json:"permission"`
	CreatedAt    time.Time `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt    time.Time `gorm:"autoUpdateTime" json:"updatedAt"`
}

// TableName handles the TableName endpoint.
func (DiskPermission) TableName() string { return "disk_permission" }

File: sdk/agentdisk/models/permission.py
"""Permission models."""

from dataclasses import dataclass


@dataclass
class DiskPermission:
    id: int
    user_id: str
    agent_id: str
    resource_id: int
    res_type: str
    permission: str
    created_at: str
    updated_at: str

    @classmethod
    def from_dict(cls, d: dict) -> "DiskPermission":
        return cls(
            id=d["id"],
            user_id=d["userId"],
            agent_id=d["agentId"],
            resource_id=d["resourceId"],
            res_type=d["resType"],
            permission=d["permission"],
            created_at=d["createdAt"],
            updated_at=d["updatedAt"],
        )


@dataclass
class PermissionCheckResponse:
    allowed: bool

    @classmethod
    def from_dict(cls, d: dict) -> "PermissionCheckResponse":
        return cls(allowed=d["allowed"])
